aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "\n",
+    "Let's first fetch our ECR image url that corresponds to the image we just built and pushed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "\n",
+    "sm = boto3.client('sagemaker')\n",
+    "ecr = boto3.client('ecr')\n",
+    "account = sess.boto_session.client('sts').get_caller_identity()['Account']\n",
+    "region = boto3.session.Session().region_name\n",
+    "\n",
+    "domain_id = 'sagemaker-studio-{}'.format(sm.list_apps()['Apps'][0]['DomainId'])\n",
+    "image_tag = ecr.list_images(repositoryName=domain_id, filter={'tagStatus':'TAGGED'})['imageIds'][0]['imageTag']\n",
+    "ecr_image = '{}.dkr.ecr.{}.amazonaws.com/{}:{}'.format(account, region, domain_id, image_tag)\n",
+    "\n",
+    "print(ecr_image)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train our model using SageMaker Training Instances"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.estimator import Estimator\n",
+    "\n",
+    "hyperparameters = {'train-steps': 100}\n",
+    "\n",
+    "instance_type = 'ml.m4.xlarge'\n",
+    "\n",
+    "estimator = Estimator(role=role,\n",
+    "                      train_instance_count=1,\n",
+    "                      train_instance_type=instance_type,\n",
+    "                      image_name=ecr_image,\n",
+    "                      hyperparameters=hyperparameters)\n",
+    "\n",
+    "estimator.fit(data_location)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Host our model using SageMaker Hosting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = estimator.deploy(1, instance_type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Making predictions using Python SDK\n",
+    "To make predictions, we use an image that is converted using OpenCV into a json format to send as an inference request. We need to install OpenCV to deserialize the image that is used to make predictions.\n",
+    "\n",
+    "The JSON reponse will be the probabilities of the image belonging to one of the 10 classes along with the most likely class the picture belongs to. The classes can be referenced from the CIFAR-10 website. Since we didn't train the model for that long, we aren't expecting very accurate results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!apt-get update\n",
+    "!apt-get install -y libsm6 libxext6 libxrender-dev libgl1-mesa-glx"
```

Comment:
Why do we need all of this? We can do image transformations w/o having to install a bunch of stuff!